### ISSUE TITLE
API deprecations on 1.16

### ISSUE THREAD

**Issue by caarlos0** [kubernetes-sigs/kubebuilder#944]
Some things were deprecated on [1.16](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16)... I'm adapting my operator to stop using those deprecated things, but I stumble across and issue with Ingresses.

On YAMLs, I would just do this:

```diff
- apiVersion: extensions/v1beta1
+ apiVersion: networking.k8s.io/v1beta1
```

But how can I do that in code?

I tried changing the imports, e.g.: 

```diff
-    extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+    networkingv1beta1 "k8s.io/api/networking/v1beta1"
```

But dep fails to resolve that. I couldn't find any examples on github as well (github search didn't help).

How can I handle this?

Sorry in advance if this is not the right place.
Thanks for the great project and all the work! 🙏

**Comment by DirectXMan12** [comment]
dep's probably failing to resolve that because either a) you've got a different version of k8s.io/api pinned, or b) something with a transitive dependency *needs* the older versions.

Which version of KubeBuilder are you using?  Which controller-runtime?

**Comment by caarlos0** [comment]
Thanks for the quick response!

> Which version of KubeBuilder are you using? Which controller-runtime?

controller-runtime v0.1.1
kubebuilder 1.0.8

**Comment by DirectXMan12** [comment]
So, we're in a bit of a weird spot with controller-runtime and 1.16 ATM.  controller-runtime v1.16 is almost guaranteed not work.  However, forcing controller-runtime `v0.2.0-rc.0` should probably work with 1.16

**Comment by DirectXMan12** [comment]
I think this can be closed now that v0.2.x is more stable
